internal/brokers/examplebroker/broker.go

```diff
@@ -124,12 +125,15 @@ func (b *Broker) GetAuthenticationModes(ctx context.Context, sessionID string, s
 	allModes := getSupportedModes(sessionInfo, supportedUILayouts)
 
 	// If the user needs mfa, we remove the last used mode from the list of available modes.
-	if sessionInfo.currentAuthStep > 1 && sessionInfo.currentAuthStep < sessionInfo.neededAuthSteps {
+	if sessionInfo.currentAuthStep > 1 && sessionInfo.currentAuthStep <= sessionInfo.neededAuthSteps {
 		allModes = getMfaModes(sessionInfo, sessionInfo.allModes)
 	}
 	// If the user needs or can reset the password, we only show those authentication modes.
 	if sessionInfo.currentAuthStep > 1 && sessionInfo.pwdChange != noReset {
```

Comment:
Actually, reading this, you are hardcoding a different way, discaring neededAuthSteps for password reset.

So, it’s either:
- you set neededAuthSteps to be 1 and find a way to return "next" on last real authentication
- change the condition to be: `if sessionInfo.pwdChange != noReset && sessionInfo.currentAuthStep == sessionInfo.neededAuthSteps {`

That way, we don’t assume "after first step, I reset the password, but I really reset the password on last step"